### PR TITLE
feat: add appProtocol and name to var.port_mappings

### PR DESCRIPTION
**Pull request by jtribble**
## what

- Adding missing fields to `var.port_mappings` (`appProtocol`, `name`, `containerPortRange`)

## why

- So that devs have access to all of the available fields: https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_PortMapping.html

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -174,7 +174,7 @@ No resources.
...
...ject({<br>      name          = optional(string)<br>      containerPort = number<br>      hostPort      = optional(number)<br>      protocol      = optional(string)<br>    })))<br>    privileged             = optional(bool)<br>    pseudoTerminal         = optional(bool)<br>    readonlyRootFilesystem = optional(bool)<br>    repositoryCredentials = optional(object({<br>      credentialsParameter = string<br>    }))<br>    resourceRequirements = optional(list(object({<br>      type  = string<br>      value = string<br>    })))<br>    secrets = optional(list(object({<br>      name      = string<br>      valueFrom = string<br>    })))<br>    startTimeout = optional(number)<br>    stopTimeout  = optional(number)<br>    systemControls = optional(list(object({<br>      namespace = string<br>      value     = string<br>    })))<br>    ulimits = optional(list(object({<br>      hardLimit = number<br>      name      = string<br>      softLimit = number<br>    })))<br>    user = optional(string)<br>    volumesFrom = optional(list(object({<br>      readOnly        = optional(bool)<br>      sourceContainer = string<br>    })))<br>    workingDirectory = optional(string)<br>  })</pre> | `{}` | no |
```
Comment (jtribble) [inline README.md:177]: Here's a more helpful view of the diff: https://www.diffchecker.com/Wrevwmek/

File: variables.tf [hunk 2]
```diff
@@ -100,10 +100,12 @@ variable "container_definition" {
...
-      name          = optional(string)
-      containerPort = number
-      hostPort      = optional(number)
-      protocol      = optional(string)
+      appProtocol        = optional(string) # http, http2, grpc (defaults to tcp)
+      containerPort      = optional(number)
+      containerPortRange = optional(string)
+      hostPort           = optional(number)
+      name               = optional(string)
+      protocol           = optional(string) # tcp (default), udp
```
Comment (jamengual) [inline variables.tf:108]: can you please reorder this and keep the same order that we have before and add the new vars after?

Comment (jtribble) [reply variables.tf:108]: Sure thing 👍

**Review by max-lobur — APPROVED** [review]

**Comment by max-lobur** [comment]
Just merged earlier PR with the same feature https://github.com/cloudposse/terraform-aws-ecs-container-definition/pull/168

**Comment by dudymas** [comment]
the other PR had the additional requirement of setting the `name` field, which is used in the ECS Service Connect api

**Comment by jtribble** [comment]
Hey @max-lobur @dudymas! #168 added `name` and `appProtocol` to `var.port_mappings`, which is the main change I needed! In this PR I also added `containerPortRange` since I noticed it was missing.

Would you like me to refactor this PR to just contain the port range change? We would also want to make `containerPort` optional since it's optional in the AWS config.

**Pull request closed without merging**